### PR TITLE
feat(react-tree): TreeItem static style for level

### PR DESCRIPTION
**Pull request by bsunderhus**
## New Behavior

1. Opts in on using static style for the first 10 levels of the tree.
2. Allows inline style override from the user side to ensure all use cases.
3. Adds stories to properly document this behavour

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/018e48731c7511dc205d74498b4c075624d2e01f'>018e48731c7511dc205d74498b4c075624d2e01f</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-components/react-tree/stories/Tree/TreeWithoutInlineStyle.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  },
+});
+
+const TreeItem = (props: TreeItemProps) => {
+  const styles = useStyles();
+  const level = useTreeContext_unstable(ctx => ctx.level);
+  return (
+    <BaseTreeItem
+      {...props}
+      style={{ [treeItemLevelToken]: undefined, ...props.style }}
```
Comment (ling1726) [inline packages/react-components/react-tree/stories/Tree/TreeWithoutInlineStyle.stories.tsx:30]:
```suggestion
      style={{ [treeItemLevelToken]: undefined }}
```

we don't need to forward `style` here right? the audience for the docs is app developers

File: packages/react-components/react-tree/stories/Tree/TreeWithoutInlineStyle.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  TreeItem as BaseTreeItem,
+  TreeItemLayout,
+  treeItemLevelToken,
+  TreeItemProps,
+  useTreeContext_unstable,
+} from '@fluentui/react-tree';
+import { makeStyles, mergeClasses } from '@fluentui/react-components';
+
+const useStyles = makeStyles({
+  level1: {
```
Comment (ling1726) [inline packages/react-components/react-tree/stories/Tree/TreeWithoutInlineStyle.stories.tsx:13]: What is we just hard coded like 10 level support and if any further levels are needed, we just show users how to add new levels... if users are going to override with their own level styling anyway what's the point?

Comment (ling1726) [reply packages/react-components/react-tree/stories/Tree/TreeWithoutInlineStyle.stories.tsx:13]: For users with lots of custom levels we can instruct them how to use the inline style method (that is currently internal) to handle that.

This way for users that won't care about more than 10 levels, it's much easier to override the styles

File: packages/react-components/react-tree/stories/Tree/TreeWithoutInlineStyle.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
+import * as React from 'react';
```
Comment (layershifter) [inline packages/react-components/react-tree/stories/Tree/TreeWithoutInlineStyle.stories.tsx:1]: This story will need a nice description, too 😉

**Review by layershifter — APPROVED** [review]

File: packages/react-components/react-tree/stories/Tree/TreeWithoutInlineStyle.stories.tsx (hunk 2 above)
Comment (bsunderhus) [reply packages/react-components/react-tree/stories/Tree/TreeWithoutInlineStyle.stories.tsx:13]: I like the idea, going with static first and documenting the downsides and how to use inline sounds like a good alternative 👍🏻

File: packages/react-components/react-tree/stories/Tree/TreeWithoutInlineStyle.stories.tsx (hunk 3 above)
Comment (bsunderhus) [reply packages/react-components/react-tree/stories/Tree/TreeWithoutInlineStyle.stories.tsx:1]: All of them need 🙈, I'll worry about documenting once I hit unstable

**Review by ling1726 — APPROVED** [review]

**Pull request merged**